### ISSUE TITLE
Are subscriptions being removed when client disconnects ?

### ISSUE THREAD

**Issue by Goluis** [apollographql/subscriptions-transport-ws#154]
Hi there, I was doing today my migration to 0.7.0, everything ended well but there was something I didn't understand why It was happening related to this code:
```js
purchaseUpdated: {
  resolve: purchase => purchase,
  subscribe: withFilter(() => pubsub.asyncIterator('purchaseUpdated'), (purchase) => {
    console.log('Hi there')
    return purchase
  })
}
```
the above code works and `hi there` is displayed in console only one time, but If then I reload the page 5 times or close the browser and again publish a message, `hi there` is displayed in console 6 times.

I tried using this:
```js
onDisconnected (webSocket) {
  webSocket.close()
  webSocket.terminate()
}
```
onDisconnected was working as expected but nothing changed.
The code I use for the migration is almost exactly the same as the one in [GitHunt-Api](https://github.com/apollographql/GitHunt-API/tree/master/api)

**Comment by dotansimha** [comment]
@Goluis 
I added tests for unsubscribe and disconnect here: https://github.com/apollographql/subscriptions-transport-ws/pull/163 
It looks like it passes (the fail is due to something else).

Can you provide a failing test or reproduction for this?

Thanks

**Comment by Goluis** [comment]
Here is a repo using Next.js: https://github.com/Goluis/with-apollo-subscriptions
Check the console for logs, keep refreshing the page and they will stack.

**Comment by marklawlor** [comment]
We're experiencing this issue as well. The websocket is being released, however the filter/resolve functions are still firing for disconnected connections.

**Comment by dotansimha** [comment]
The actual issue was with `graphlql-subscriptions` (`withFilter`), it was fixed by @kouak (https://github.com/apollographql/graphql-subscriptions/pull/74). 
We will release a new version soon!

**Comment by Urigo** [comment]
released in `0.4.3`
